### ISSUE TITLE
nodetool removenode fails for CDC tables

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylla#8934]
*Installation details*
Scylla version (or git commit hash): 4.5.rc3-0.20210620.706de00ef with build-id d28ca2bf35abaa8bbf9a2b5b401a0bbe20ad506e
Cluster size: 6
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0945c02deff78e980 (eu-west-1)

test name is `longevity-cdc-3d-400gb` and after around a day it ran namesis `disrupt_remove_node_then_add_node`, that will:
1. stop Scylla in a target node
2. terminate the node
3. run full `nodetool repair` in all nodes that stayed alive (best effort)
4. run `nodetool removenode`
5. add a new node to the cluster
6. run full repair

we failed after step 3, while running removenode (of the dead node):
```
< t:2021-06-24 20:26:30,722 f:base.py l:220 c:RemoteCmdRunner p:DEBUG > nodetool: Scylla API server HTTP POST to URL '/storage_service/remove_node' failed: std::runtime_error ({shard 0: seastar::rpc::closed_error (connection is closed), shard 1: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 2: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 3: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 4: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 5: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 6: seastar::rpc::closed_error (connection is closed), shard 7: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 8: seastar::rpc::closed_error (connection is closed), shard 9: seastar::rpc::closed_error (connection is closed), shard 10: seastar::rpc::stream_closed (rpc stream was closed by peer), shard 11: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 12: seastar::rpc::closed_error (connection is closed), shard 13: seastar::rpc::closed_error (connection is closed)})
```

full logs can be found here:
[sct-runner.logs](https://cloudius-jenkins-test.s3.amazonaws.com/80a8ac7f-e090-4e64-8876-6779a113ba02/20210624_211439/sct-runner-80a8ac7f.tar.gz)
[db-cluster.logs](https://cloudius-jenkins-test.s3.amazonaws.com/80a8ac7f-e090-4e64-8876-6779a113ba02/20210624_211439/db-cluster-80a8ac7f.tar.gz)

`test-id: 80a8ac7f-e090-4e64-8876-6779a113ba02`

**Comment by fgelcer** [comment]
@roydahan reminded me that this failure is "expected" as this repair is a best effort for the "situation"

there is a failure though during the next command, `nodetool removenode`:
```
< t:2021-06-24 20:26:30,722 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > nodetool: Scylla API server HTTP POST to URL '/storage_service/remove_node' failed: std::runtime_error ({shard 0: seastar::rpc::closed_error (connection is closed), shard 1: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 2: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 3: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 4: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 5: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 6: seastar::rpc::closed_error (connection is closed), shard 7: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_preimage, range=(-9089371111717512258, -9063739181969075388]), shard 8: seastar::rpc::closed_error (connection is closed), shard 9: seastar::rpc::closed_error (connection is closed), shard 10: seastar::rpc::stream_closed (rpc stream was closed by peer), shard 11: std::runtime_error (Failed to repair for keyspace=cdc_test, cf=test_table_scylla_cdc_log, range=(-9089371111717512258, -9063739181969075388]), shard 12: seastar::rpc::closed_error (connection is closed), shard 13: seastar::rpc::closed_error (connection is closed)})
```

**Comment by fgelcer** [comment]
@roydahan , after that failure during `nodetool removenode` we see in the logs:
```
< t:2021-06-24 20:36:43,533 f:nemesis.py      l:296  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: {'start': 1624565183, 'end': 1624566396, 'duration': 1212, 'node': 'Node longevity-cdc-3d-4
00gb-4-5-db-node-80a8ac7f-2 [34.248.95.142 | 10.0.0.72] (seed: False)', 'subtype': 'end', 'error': "Node was not removed properly (Node status:{'state': 'DN', 'load': '491.34GB', 'tokens': '256', 'owns':
 '?', 'host_id': '9ef6b2c6-b12e-4214-89c9-e1ec963cafd0', 'rack': '1a'})", 'full_traceback': 'Traceback (most recent call last):\n  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3002, in 
wrapper\n    result = method(*args, **kwargs)\n  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3306, in disrupt\n    self.call_random_disrupt_method()\n  File "/home/ubuntu/scylla-cluste
r-tests/sdcm/nemesis.py", line 1144, in call_random_disrupt_method\n    self.execute_disrupt_method(disrupt_method)\n  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1151, in execute_disr
upt_method\n    disrupt_method()\n  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 2369, in disrupt_remove_node_then_add_node\n    assert removed_node_status is None,\\\nAssertionError: N
ode was not removed properly (Node status:{\'state\': \'DN\', \'load\': \'491.34GB\', \'tokens\': \'256\', \'owns\': \'?\', \'host_id\': \'9ef6b2c6-b12e-4214-89c9-e1ec963cafd0\', \'rack\': \'1a\'})\n'}
```

and the nemesis has an `assert`:
```
assert removed_node_status is None,\
                "Node was not removed properly (Node status:{})".format(removed_node_status)
```

where exactly we see the last message from the nemesis, before:
```
< t:2021-06-24 20:36:43,663 f:tester.py       l:187  c:silence              p:DEBUG > Finished 'get_nemesis_report(cluster=Cluster longevity-cdc-3d-400gb-4-5-db-cluster-80a8ac7f (AMI: ['ami-0945c02deff78e980'] Type: i3.4xlarge))'. No errors were silenced.
< t:2021-06-24 20:36:43,741 f:tester.py       l:176  c:silence              p:DEBUG > Silently running 'stop_nemesis'
```

**Comment by fgelcer** [comment]
before we ran the above `assert`, this is what we ran:
```
removed_node_status = self.cluster.get_node_status_dictionary(
                ip_address=node_to_remove.ip_address, verification_node=verification_node)
```
and its result is:
```
< t:2021-06-24 20:26:32,977 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > ---
< t:2021-06-24 20:26:34,214 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > Datacenter: eu-west
< t:2021-06-24 20:26:34,435 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > ===================
< t:2021-06-24 20:26:34,815 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > Status=Up/Down
< t:2021-06-24 20:26:34,890 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2021-06-24 20:26:34,909 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > --  Address     Load       Tokens       Owns    Host ID                               Rack
< t:2021-06-24 20:26:35,122 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > UN  10.0.3.1    614.26 GB  256          ?       f41ad1b3-5baf-4700-b1f3-7063e4a71175  1a
< t:2021-06-24 20:26:35,229 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > UN  10.0.0.134  338.95 GB  256          ?       e62b68fd-ae8d-40ef-83aa-89579099a8bc  1a
< t:2021-06-24 20:26:35,372 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.215  459.32 GB  256          ?       c0b2bf5b-82df-442e-9479-c0dd5d0943b8  1a
< t:2021-06-24 20:26:35,478 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > DN  10.0.0.72   491.34 GB  256          ?       9ef6b2c6-b12e-4214-89c9-e1ec963cafd0  1a
< t:2021-06-24 20:26:35,504 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.204  438 GB     256          ?       0e1320bc-5266-4462-88f5-785caf594aad  1a
< t:2021-06-24 20:26:35,650 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > UN  10.0.2.28   479.97 GB  256          ?       09e97154-19bc-46f4-9391-3d5c469f7292  1a
< t:2021-06-24 20:26:35,651 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > 
< t:2021-06-24 20:26:35,797 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > Note: Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless
```

**Comment by fgelcer** [comment]
so the issue in here is that `nodetool remove` failed

**Comment by fgelcer** [comment]
could be the same as #7965 (but the error message from `nodetool` command has more information

**Comment by haaawk** [comment]
The question is whether this is CDC related or could happen to any table? CDC table is a regular table so I would expect that the issue is not CDC related.

**Comment by roydahan** [comment]
After connecting this cluster manually and running the same "nodetool removenode" command manually (while there is no load running on this cluster) the command succeed and the node was removed.

It may be related to another issue that repair with TWCS fails or something like that (need to find the issue).

**Comment by slivne** [comment]
@asias can you please help here
- need to validate that the repair ended prior to removenode starting 
- according to QA they start the removenode operation only after the nodetool repair ends - I am suspecting that the command may retrn and there is a background non completed repair still running.

**Comment by asias** [comment]
The nodetool removenode was performed on node 6. 

```
$ grep -r 'Started to sync data for removing' longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-*
longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-1/messages.log:2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-1 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started to sync data for removing node=10.0.0.72, coordinator=10.0.3.1
longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-3/messages.log:2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-3 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started to sync data for removing node=10.0.0.72, coordinator=10.0.3.1
longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6/messages.log:2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started to sync data for removing node=10.0.0.72, coordinator=10.0.3.1
longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-7/messages.log:2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-7 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started to sync data for removing node=10.0.0.72, coordinator=10.0.3.1
longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-8/messages.log:2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-8 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started to sync data for removing node=10.0.0.72, coordinator=10.0.3.1
```


The version does not have

````
commit 3896e358972ced18c60654ffaa2b1f260f67fd5f
Merge: e9c940dbbc 70147dcb5a
Author: Avi Kivity <avi@scylladb.com>
Date:   Tue May 25 18:44:46 2021 +0300

    Merge 'storage_service: Respect --enable-repair-based-node-ops flag during removenode' from Asias He
    
    In commit 829b4c1 (repair: Make removenode safe by default), removenode
    was changed to use repair based node operations unconditionally. Since
    repair based node operations is not enabled by default, we should
    respect the flag to use stream to sync data if the flag is false.
    
    Fixes #8700
    
    Closes #8701
    
    * github.com:scylladb/scylla:
      storage_service: Add removenode_add_ranges helper
      storage_service: Respect --enable-repair-based-node-ops flag during removenode
```

So, RBNO was enabled for removenode unconditionally.

**Comment by asias** [comment]
Before removenode was performed, _read_concurrency_sem timeout error was observed. 
```
197506 2021-06-24T20:08:46+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !ERR     | scylla:  [shard 13] storage_proxy - Exception when communicating with 10.0.3.1, to read from cdc_test.test_table: seastar::named_semaphore_t       imed_out (Semaphore timed out: _read_concurrency_sem)
197507 2021-06-24T20:08:46+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !ERR     | scylla: message repeated 3 times: [  [shard 13] storage_proxy - Exception when communicating with 10.0.3.1, to read from cdc_test.test_table       : seastar::named_semaphore_timed_out (Semaphore timed out: _read_concurrency_sem)]
197508 2021-06-24T20:08:46+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !ERR     | scylla:  [shard 13] storage_proxy - Exception when communicating with 10.0.3.1, to read from cdc_test.test_table_preimage: seastar::named_se       maphore_timed_out (Semaphore timed out: _read_concurrency_sem)
197509 2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Started removenode operation, removing node=10.0.0.72       , sync_nodes={10.0.3.1, 10.0.2.28, 10.0.0.134, 10.0.2.204, 10.0.2.215}, ignore_nodes=[]
197510 2021-06-24T20:08:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] storage_service - removenode[446f4672-0f1d-44ab-b532-50c68627c52b]: Added node=10.0.0.72 as leaving node, coordinator=10.       0.3.1
```

**Comment by asias** [comment]
@fgelcer 

Here is all repair performed on node 6.

```
[asias@hjpc2 db-cluster-80a8ac7f]$ cat 6|grep user-requested|tail
2021-06-24T13:34:32+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1196, uuid=93b847b0-8608-4605-9ec8-0a55d60940bf], options {{ hosts -> 10.0.0.213,10.0.0.72,10.0.3.1}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 4257185922974093637:4267988679188513929}}
2021-06-24T13:34:32+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1197, uuid=60c0160e-d2dd-4390-b220-c46809538949], options {{ hosts -> 10.0.3.1,10.0.0.213,10.0.0.72}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3171878920572289148:3173639860206364525}}
2021-06-24T13:34:33+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1198, uuid=9984c7c1-654c-4abe-ac0c-9a68cd1da13a], options {{ hosts -> 10.0.0.72,10.0.3.1,10.0.0.213}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 6036527554995086591:6045792814702598862}}
2021-06-24T13:34:35+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1199, uuid=53f82d77-e565-4603-bd67-f9350b21196f], options {{ hosts -> 10.0.0.213,10.0.0.72,10.0.3.1}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3173639860206364525:3193154821746600574}}
2021-06-24T13:34:37+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1200, uuid=47a73b93-aef1-4713-8300-8fae2155cbc8], options {{ hosts -> 10.0.3.1,10.0.0.213,10.0.0.72}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3688001777043859743:3699795980529986034}}
2021-06-24T13:34:37+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1201, uuid=f1bf4002-d3b5-4307-84d3-29554b7f2f42], options {{ hosts -> 10.0.0.213,10.0.0.72,10.0.3.1}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3548582235813258220:3550739217480892663}}
2021-06-24T13:34:37+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1202, uuid=05865fa7-f73c-4f00-9bb7-47e3c35addcf], options {{ hosts -> 10.0.3.1,10.0.0.213,10.0.0.72}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3593210207239951866:3598500770028777903}}
2021-06-24T13:34:39+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1203, uuid=640ede11-3cc1-4c8f-9188-820beda12a2d], options {{ hosts -> 10.0.0.213,10.0.3.1,10.0.0.72}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> 3598500770028777903:3601963242991880512}}
2021-06-24T13:34:39+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace cdc_test, repair id [id=1204, uuid=08ce218f-ee71-46ae-b261-972ddb0dfc58], options {{ hosts -> 10.0.3.1,10.0.0.213,10.0.0.72}, { columnFamilies -> test_table_preimage_postimage}, { ranges -> -2425213951147607875:-2405742948941644736}}
2021-06-24T20:08:25+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace system_auth, repair id [id=1, uuid=d41201f5-68b3-4df5-be44-cb84ade42320], options {{ trace -> false}, { jobThreads -> 1}, { incremental -> false}, { parallelism -> parallel}, { primaryRange -> false}}
```


The last one with id 1024, uuid 08ce218f-ee71-46ae-b261-972ddb0dfc58 finished successfully.

```
 66735 2021-06-24T13:34:43+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - repair id [id=1204, uuid=08ce218f-ee71-46ae-b261-972ddb0dfc58] completed successfull
```

Was those repair performed for Step 3, repair before removenode? Those repairs were way earlier (2021-06-24T13:34:43+) than the removenode operations which started at `2021-06-24T20:08:49`.

**Comment by asias** [comment]
After repair for removenode started for cdc_test started, we can see nodes 10.0.2.28 (node7) and 10.0.2.204 (node1) were marked as done by 10.0.3.1 (node 6). Node 1 and node 7 were part of the removenode operations (See https://github.com/scylladb/scylla/issues/8934#issuecomment-871868599).

The reason for the failure of the repair for removenode was the participants nodes node 1 and node7 were marked as down which failed the removenode ops.

We need to backport the following to help avoiding marking node down during high workload. 

```
commit 9ea57dff212006ab6b9ee836c9e25ec53a0e702a
Author: Asias He <asias@scylladb.com>
Date:   Tue Apr 13 21:37:12 2021 +0800

    gossip: Relax failure detector update
    
    We currently only update the failure detector for a node when a higher
    version of application state is received. Since gossip syn messages do
    not contain application state, so this means we do not update the
    failure detector upon receiving gossip syn messages, even if a message
    from peer node is received which implies the peer node is alive.
    
    This patch relaxes the failure detector update rule to update the
    failure detector for the sender of gossip messages directly.
    
    Refs #8296
    
    Closes #8476

```

A even better commit is the new failure detector but back-porting it to 4.5 is not realistic. 

```
commit 425e3b11827ab5db6d68024134233e85064c7f74
Author: Asias He <asias@scylladb.com>
Date:   Thu Feb 4 10:29:49 2021 +0800

    gossip: Introduce direct failure detector

```




```
2021-06-24T20:14:02+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] repair - repair id [id=4, uuid=f4f685c4-71c9-4edc-87b6-6583473b389d] to sync data for keyspace=cdc_test, status=started
```

```
[asias@hjpc2 db-cluster-80a8ac7f]$ cat 6.after.repair.for.removenode|grep DOWN
2021-06-24T20:17:20+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:18:50+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:19:26+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:20:07+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:20:42+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:21:44+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:22:50+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:24:04+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:24:44+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:25:13+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:25:51+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:26:06+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.204 is now DOWN, status = NORMAL
2021-06-24T20:26:52+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:27:31+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:28:06+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:28:40+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.204 is now DOWN, status = NORMAL
2021-06-24T20:28:41+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:29:15+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:29:50+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:30:19+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:30:32+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:30:38+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.204 is now DOWN, status = NORMAL
2021-06-24T20:31:38+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:33:14+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:33:43+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:33:47+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:35:10+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:35:13+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:36:02+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:36:03+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:37:49+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
2021-06-24T20:40:01+00:00  longevity-cdc-3d-400gb-4-5-db-node-80a8ac7f-6 !INFO    | scylla:  [shard 0] gossip - InetAddress 10.0.2.28 is now DOWN, status = NORMAL
```

**Comment by slivne** [comment]
@fgelcer / @roydahan is this a regression ? did we have the same issue in 4.4 ?

**Comment by fgelcer** [comment]
> @fgelcer / @roydahan is this a regression ? did we have the same issue in 4.4 ?

searching for 4.4 issues related to CDC and repair i found 2 issues, but they are not the same (it could be that, we have different message for them, but it might not be the same):
#7570 and #8296

**Comment by slivne** [comment]
@asias can you check if the issues fabio found related to 4.4 and removenode with CDC are they related or not - we need to understand if we have a regression at hand and fix it / do a bisect and find the source

**Comment by slivne** [comment]
@asias / @bhalevy ping

**Comment by slivne** [comment]
@asias / @bhalevy ping

**Comment by asias** [comment]
> > @fgelcer / @roydahan is this a regression ? did we have the same issue in 4.4 ?
> 
> searching for 4.4 issues related to CDC and repair i found 2 issues, but they are not the same (it could be that, we have different message for them, but it might not be the same):
> #7570 and #8296

The are all very similar. All due to a node is being marked as down during repair + cdc. We have fixed in the commits I mentioned here:

https://github.com/scylladb/scylla/issues/8934#issuecomment-871876915

**Comment by slivne** [comment]
According to Asias this is not a regression - moving to 4.6

**Comment by slivne** [comment]
To make it clear this should pass on 4.6 (master) that has Asias patch please confirm

**Comment by bhalevy** [comment]
@fgelcer I assigned this issue to you, please confirm the test passes in 4.6 as per https://github.com/scylladb/scylla/issues/8934#issuecomment-882460044

Cc @roydahan

**Comment by fgelcer** [comment]
> @fgelcer I assigned this issue to you, please confirm the test passes in 4.6 as per [#8934 (comment)](https://github.com/scylladb/scylla/issues/8934#issuecomment-882460044)
> 
> Cc @roydahan

running a reproducer (on the cdc-4h job) with only the nemesis that detected this failure, [here](https://jenkins.scylladb.com/view/master/job/scylla-master/job/reproducers/job/reproducer_8934_cdc_4h/)

**Comment by fgelcer** [comment]
> > @fgelcer I assigned this issue to you, please confirm the test passes in 4.6 as per [#8934 (comment)](https://github.com/scylladb/scylla/issues/8934#issuecomment-882460044)
> > Cc @roydahan
> 
> running a reproducer (on the cdc-4h job) with only the nemesis that detected this failure, [here](https://jenkins.scylladb.com/view/master/job/scylla-master/job/reproducers/job/reproducer_8934_cdc_4h/)

@bhalevy , in my reproducer, i got the nemesis RemoveNodeThenAddNode to run 6 times, and all of them passed (Scylla version 4.6.dev-0.20210811.e5bb88b69 with build-id 210fb1f13e1241d4b824142cbffa64bcffdcf1af):
```
2021-08-12 15:12:58.893: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=2501826b-b663-42c8-a907-7192caffe9f6: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-5 [54.246.59.241 | 10.0.3.17] (seed: False)
2021-08-12 15:38:23.952: (DisruptionEvent Severity.NORMAL) period_type=end event_id=2501826b-b663-42c8-a907-7192caffe9f6 duration=25m25s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-5 [54.246.59.241 | 10.0.3.17] (seed: False)
2021-08-12 15:46:06.498: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=378feddf-0d36-4aef-b4ad-5d5f9bafb447: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-6 [63.32.44.230 | 10.0.1.154] (seed: False)
2021-08-12 16:22:10.000: (DisruptionEvent Severity.NORMAL) period_type=end event_id=378feddf-0d36-4aef-b4ad-5d5f9bafb447 duration=36m3s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-6 [63.32.44.230 | 10.0.1.154] (seed: False)
2021-08-12 16:29:49.923: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=9e425a3b-1e58-4440-8da5-3702260c6370: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-3 [34.240.224.210 | 10.0.3.153] (seed: False)
2021-08-12 17:17:34.942: (DisruptionEvent Severity.NORMAL) period_type=end event_id=9e425a3b-1e58-4440-8da5-3702260c6370 duration=47m45s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-3 [34.240.224.210 | 10.0.3.153] (seed: False)
2021-08-12 17:24:49.081: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=a638f6c3-cc0f-47da-a042-404fc8d09e3f: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-8 [54.247.52.33 | 10.0.0.45] (seed: False)
2021-08-12 18:13:07.061: (DisruptionEvent Severity.NORMAL) period_type=end event_id=a638f6c3-cc0f-47da-a042-404fc8d09e3f duration=48m17s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-8 [54.247.52.33 | 10.0.0.45] (seed: False)
2021-08-12 18:21:11.421: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=d1f4e7ca-e487-4e1a-919f-0675dc1a624a: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-4 [54.75.40.115 | 10.0.0.247] (seed: False)
2021-08-12 18:49:12.123: (DisruptionEvent Severity.NORMAL) period_type=end event_id=d1f4e7ca-e487-4e1a-919f-0675dc1a624a duration=28m0s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-4 [54.75.40.115 | 10.0.0.247] (seed: False)
2021-08-12 18:55:35.990: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=46941492-eed7-4394-9739-eb267727a456: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-10 [54.229.37.201 | 10.0.0.83] (seed: False)
2021-08-12 19:15:20.706: (DisruptionEvent Severity.NORMAL) period_type=end event_id=46941492-eed7-4394-9739-eb267727a456 duration=19m44s: nemesis_name=RemoveNodeThenAddNode target_node=Node longevity-cdc-100gb-4h-check-fi-db-node-37f6281d-10 [54.229.37.201 | 10.0.0.83] (seed: False)
```

**Comment by fgelcer** [comment]
> To make it clear this should pass on 4.6 (master) that has Asias patch please confirm

@slivne @bhalevy , it seems like Asias patch works, as in [this comment](https://github.com/scylladb/scylla/issues/8934#issuecomment-899037597)
